icu4c/source/i18n/number_mapper.h

```diff
@@ -136,7 +136,7 @@ struct DecimalFormatFields : public UMemory {
     * The pre-computed formatter object. Setters cause this to be re-computed atomically. The {@link
     * #format} method uses the formatter directly without needing to synchronize.
     */
-    LocalPointer<const LocalizedNumberFormatter> formatter;
+    LocalPointer<LocalizedNumberFormatter> formatter;
```

Comment:
Note: We need to remove the `const` here so that we can move assign to it in the `touch` function.